modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -1539,6 +1542,85 @@ public void deploysAppZipWithExtensionXmlPlugin() throws Exception {
   }
 
   @Test
+  public void deploysWithExtensionXmlPluginWithPomWithoutMulePluginDependencies() throws Exception {
```

Comment:
I think that only one test must cover deployment of plugins using the json, the other ones look like tests on ArtifactPluginDescriptorFactory, once you test that the created descriptor is the right one, there is no need to deploy a full application